src/interfaces/theme.ts

```diff
@@ -40,4 +40,41 @@ export const STANDARD_CHALK = [
 
 export type StandardChalk = (typeof STANDARD_CHALK)[number]
 
-export type Theme = Record<string, Color | StandardChalk>
+export const THEME_KEYS = [
+  'alias',
+  'bin',
+  'command',
+  'commandSummary',
+  'dollarSign',
+  'flag',
+  'flagDefaultValue',
+  'flagOptions',
+  'flagRequired',
+  'flagSeparator',
+  'flagType',
+  'sectionDescription',
+  'sectionHeader',
+  'topic',
+  'version',
+] as const
+
+export type ThemeKey = (typeof THEME_KEYS)[number]
+
+export type Theme = {
+  [key: string | ThemeKey]: Color | StandardChalk | undefined
```

Comment:
Ok, I see a use case for this. CLI developers can extend the oclif's Theme Spec with their own theme specific keys so that plugins built with their own CLI API can take advantage of their own Theme Spec, which is just a combination of Oclif + CLI Theme Specs. Makes sense now.

Reply:
Shouldn't Theme be a `interface` to allow it to be extensible in typescript world?